live-site/components/coming-up/ComingUp.tsx

```diff
@@ -10,31 +10,38 @@ import {
   StyledTextContainer,
   StyledSectionHeader,
   StyledEvents,
-  StyledOneEventContainer
+  StyledOneEventContainer,
+  StyledLoadingText
 } from './ComingUp.styles';
 import { min } from '../../../shared-ui/lib/responsive';
 import useMatchMedia from 'react-use-match-media';
 import { useAirtableApi } from '../../src/hooks/useAirtable';
 
 const ComingUpSection: React.FC = () => {
-  const { data } = useAirtableApi('Relevant', 'relevant', true);
-  let count = 0; 
-  let events: UpcomingEvent[]= Array.from(
-    new Set(
-    data.map((event) => {
-      count = count + 1;
-        return {
-          id: count,
-          header: event.fields.title,
-          time: event.fields.start_time,
-          display_start_time: event.fields.display_start_time,
-          body: event.fields.notes,
-        } 
-    }))
-  )
-  events.sort((event1: UpcomingEvent, event2: UpcomingEvent) => (event1.time > event2.time) ? 1 : -1);
-  events = events.slice(0, 3); 
+  const { data, isLoading } = useAirtableApi('Relevant', 'relevant', true);
   const isDesktop = useMatchMedia(min.tablet);
+  
+  let count = 0;
+  let events: UpcomingEvent[] = data.map((event) => {
```

Comment:
I think we can put this chunk of code in a `useEffect` - that way, we'll only run it when `data` changes, rather than running it on every render.